### DISCUSSION TITLE
Can I implement an automatic analyzer in a plugin?

### DISCUSSION THREAD

**Discussion by Forsworns** [NationalSecurityAgency/ghidra#6235] [Q&A] [answered]
Recently I wrote a [plugin](https://github.com/Forsworns/GhidrAI) to request LLM services. The GUI tools work well, but I cannot find my analyzer inside the list of `Analysis > Auto Analyze`. Do I need explicitly register my analyzer with an API similar to the `PluginInfo`? I noticed that the native analyzers only need to extend `AbstractAnalyzer`.

I found a plugin [ghidra-findcrypt](https://github.com/TorgoTorgo/ghidra-findcrypt/) which implement its analyzer. But I still don't know how to get my anaylzer registered to the `Analysis > Auto Analyze`.

Here is my anaylzer:

https://github.com/Forsworns/GhidrAI/blob/main/src/main/java/ghidrai/GhidraiAnaylzer.java

**Comment by astrelsky** [comment, marked as answer]
ExtensionPoints **MUST** follow naming rules for automatic discovery. Rename `GhidraiAnaylzer` to `GhidraiAnalyzer`.

I looked at the name several times very carefully and did not see the typo the first couple times even when I was looking for it.

**Reply by Forsworns** [reply]
Thanks, sorry for my typo :(
